src/openlifu/db/database.py

```diff
@@ -107,6 +107,34 @@ def write_session(self, subject:Subject, session:Session, on_conflict=OnConflict
 
         self.logger.info(f"Added session with ID {session_id} for subject {subject.id} to the database.")
 
+    def write_run(self, run:Run, session:Session = None, protocol:Protocol = None, on_conflict=OnConflictOpts.ERROR):
+        # Check whether the run already exist in the session
+        run_ids = self.get_run_ids(session.subject_id, session.id)
+
+        if run.id in run_ids:
+            if on_conflict == OnConflictOpts.ERROR:
+                raise ValueError(f"Run with ID {run.id} already exists for session {session.id}.")
+            elif on_conflict == OnConflictOpts.OVERWRITE:
+                raise ValueError("OVERWRITE is not supported for write_run")
```

Comment:
```suggestion
                raise ValueError("Runs are write-once objects and may not be overwritten.")
```

(just to make the intention clear -- it's not just not supported but it's deliberately disallowed)